src/Sulu/Bundle/AdminBundle/Resources/js/components/Modal/tests/ClickModal.test.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable flowtype/require-valid-file-annotation */
+import ClickModal from '../ClickModal';
+import React from 'react';
+import {mount} from 'enzyme';
+import pretty from 'pretty';
+
+afterEach(() => document.body.innerHTML = '');
+
+test('The modal should initially not be rendered', () => {
+    const body = document.body;
+    const view = mount(
+        <ClickModal clickElement={<button>Open modal</button>}>
+            <p>Modal content</p>
+        </ClickModal>
+    ).render();
+    expect(view).toMatchSnapshot();
+    expect(body.innerHTML).toBe('');
+});
+
+test('The modal should be rendered when the button got clicked', () => {
+    const body = document.body;
+    const view = mount(
+        <ClickModal clickElement={<button>Open modal</button>}>
+            <p>Modal content</p>
+        </ClickModal>
+    );
+    view.render();
+
+    expect(body.innerHTML).toBe('');
+    view.find('button').simulate('click');
+    expect(pretty(body.innerHTML)).toMatchSnapshot();
```

Comment:
As mentioned, the only thing in the body is the stuff of the portal and that's exactly the things a snapshot should be taken of.